core/src/main/java/aima/core/search/api/SearchForActionsBidirectionallyFunction.java

```diff
@@ -1,12 +1,14 @@
 package aima.core.search.api;
 
-import java.util.function.BiFunction;
+import aima.core.util.datastructure.Pair;
+
+import java.util.List;
+import java.util.function.Function;
 
 /**
  * @author manthan
  */
 @FunctionalInterface
-public interface SearchForActionsBidirectionallyFunction<A, S> extends BiFunction<Problem<A, S>, Problem<A, S>, BidirectionalActions<A>> {
-    @Override
-    BidirectionalActions<A> apply(Problem<A, S> originalProblem, Problem<A, S> reverseProblem);
+public interface SearchForActionsBidirectionallyFunction<A, S> extends Function<Pair<Problem<A,
+    S>, Problem<A, S>>, List<A>> {
```

Comment:
> I am not sure of the advantage of switching from extending BiFunction to Function

there is no real benefit, I can revert that without problems.

> route finding application that gives directions to two people to meet in the middle

ok, if you find that useful, I won't argue. I'd still rename `BidirectionalActions` to something like `BiDirectionalSearchSolution` because the name is pretty misleading imho